addons/mail/static/src/widgets/form_renderer/form_renderer.js

```diff
@@ -153,6 +153,10 @@ FormRenderer.include({
             } else {
                 await this._updateChatterContainerComponent();
             }
+            let session = this.getSession();
+            // get the currency of the record if there is one
+            const currencyIdVariable = Object.entries(this.state.fields).find(field => field[1].relation === 'res.currency');
+            session.record_currency_id = currencyIdVariable && this.state.data[currencyIdVariable[0]] && this.state.data[currencyIdVariable[0]].res_id || false;
```

Comment:
I don't think the form renderer should put stuff in the session.

But anyway, this raises an important question.
Doesn't this show the **current** value of the currency for the record?
Meaning that if I change from EURO to DOLLAR multiple times and change a tracked monetary field along the way, it will show the current currency value for ALL tracking messages, instead of the currency value at the time the monetary field was changed?